src/index.ts

```diff
@@ -5,16 +5,21 @@ export class PromiseQueueError extends TypedError {}
 export class MaxSizeExceededError extends PromiseQueueError {}
 export class TimeoutError extends PromiseQueueError {}
 
-const durationSince = (t0: ReturnType<typeof process.hrtime>): number => {
-	const diff = process.hrtime(t0);
-	return diff[0] * 1000 + diff[1] / 1e6;
+// duration in bigint nanoseconds since t0
+const durationSince = (t0: bigint): bigint => {
+	return process.hrtime.bigint() - t0;
+};
+
+// convert bigint nanoseconds to number seconds
+const toSeconds = (t: bigint): number => {
+	return Number(t / BigInt(1e6)) / 1e3;
```

Comment:
Is there a reason for the two separate divisions?

Reply:
I was thinking it would be useful in future to have milliseconds, but I realize just now that this is not necessarily useful at all, such a future is likely never to occur, and the extra division, when occurring as frequently as this code path is traveled, will be an issue. It'll come out!